OpenRA.Mods.Common/ServerTraits/LobbyCommands.cs

```diff
@@ -750,6 +765,46 @@ static bool Team(S server, Connection conn, Session.Client client, string s)
 			return true;
 		}
 
+		static bool ToggleSpawn(S server, Connection conn, Session.Client client, string s)
```

Comment:
IMO it is still annoyingly awkward to go straight from an active spawn containing a player to disabled. It would be much more usable if we could refocus this as a "clear spawn" feature, where clearing an active spawn removes the player, and clearing an empty spawn disables it.

Something like this:
```csharp
static bool ClearSpawn(S server, Connection conn, Session.Client client, string s)
{
	var spawnPoint = Exts.ParseIntegerInvariant(s);
	if (spawnPoint == 0)
		return true;

	var existingClient = server.LobbyInfo.Clients.FirstOrDefault(cc => cc.SpawnPoint == spawnPoint);
	if (client != existingClient && !client.IsAdmin)
	{
		server.SendOrderTo(conn, "Message", "Only admins can clear spawn points.");
		return true;
	}

	// Clearing a selected spawn point removes the player
	if (existingClient != null)
	{
		existingClient.SpawnPoint = 0;
		server.SyncLobbyClients();
		return true;
	}

	// Clearing an empty spawn point prevents it from being selected
	// Clearing a disabled spawn restores it for use
	if (!server.LobbyInfo.DisabledSpawns.Contains(spawnPoint))
		server.LobbyInfo.DisabledSpawns.Add(spawnPoint);
	else
		server.LobbyInfo.DisabledSpawns.Remove(spawnPoint);

	server.SyncLobbyInfo();
	return true;
}
```

and then in LobbyUtils.cs:
```csharp
static void ClearPlayerSpawnPoint(OrderManager orderManager, MapPreviewWidget mapPreview, MapPreview preview, MouseInput mi)
{
	var selectedSpawn = DetermineSelectedSpawn(mapPreview, preview, mi);
	if (Game.IsHost || orderManager.LobbyInfo.Clients.FirstOrDefault(cc => cc.SpawnPoint == selectedSpawn) == orderManager.LocalClient)
		orderManager.IssueOrder(Order.Command("clear_spawn {0}".F(selectedSpawn)));
}
```

Reply:
I removed the `Game.IsHost` check in the `SelectSpawnPoint` method, following from your logic in the `ClearPlayerSpawn` method, this allows players to vacate a spawn using RMB themselves. The alert of insufficient spawns and preventing disabling too many is removed, but that solves the awkward wording issue, which is handled by the disabled Start Game button.